stdlib/encodings/utf_16.pyi

```diff
@@ -0,0 +1,29 @@
+import codecs
+from typing import IO
+
+class IncrementalEncoder(codecs.IncrementalEncoder):
+    def __init__(self, errors: str = ...) -> None: ...
+    def encode(self, input: str, final: bool = ...) -> bytes: ...
+    def reset(self) -> None: ...
+    def getstate(self) -> int: ...  # type: ignore[override]
+    def setstate(self, state: int) -> None: ...  # type: ignore[override]
+
+class IncrementalDecoder(codecs.BufferedIncrementalDecoder):
+    def __init__(self, errors: str = ...) -> None: ...
+    def _buffer_decode(self, input: bytes, errors: str | None, final: bool) -> tuple[str, int]: ...
+    def reset(self) -> None: ...
+    def getstate(self) -> tuple[bytes, int]: ...
+    def setstate(self, state: tuple[bytes, int]) -> None: ...
+
+class StreamWriter(codecs.StreamWriter):
+    def __init__(self, stream: IO[bytes], errors: str = ...) -> None: ...
+    def reset(self) -> None: ...
+    def encode(self, input: str, errors: str | None = ...) -> tuple[bytes, int]: ...
+
+class StreamReader(codecs.StreamReader):
+    def reset(self) -> None: ...
+    def decode(self, input: bytes, errors: str | None = ...) -> tuple[str, int]: ...
+
+def getregentry() -> codecs.CodecInfo: ...
+def encode(__str: str, __errors: str | None = ..., __byteorder: int = ...) -> tuple[bytes, int]: ...
```

Comment:
Quick question here for my understanding as this is my first time typing from one of the C functions:

I know the parameters/types for `codecs.utf_16_encode` by looking at the C code [here](https://github.com/python/cpython/blob/078abb676cf759b1e960f78390b6e80f256f0255/Modules/_codecsmodule.c#L703).

Looking at the similar typing in `utf_8.py` I've prefaced the parameters with a `__` to be consistent. Is that standard practice for all of the C functions?

Reply:
`__str` means that the argument is positional-only, meaning that you can't do this:

```
>>> from encodings.utf_16 import encode
>>> encode(str="lol")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: _codecs.utf_16_encode() takes no keyword arguments
```